### ISSUE TITLE
`new_data_frame()` signature collisions

### ISSUE THREAD

**Issue by DavisVaughan** [r-lib/vctrs#949]
See https://github.com/tidyverse/dplyr/issues/5042

The `n` argument in `new_data_frame()` is causing issues because you can't set an attribute named `n`. The arguments should all probably be prefixed with a `.` to prevent other collisions.

@lionel- has mentioned in the past that `n` should probably be `size`, so this seems as good a time as any to change it to:

```
new_data_frame(.x = list(), .size = NULL, ..., .class = NULL)
```

**Comment by hadley** [comment]
Should we put `.size` after `...`?

**Comment by DavisVaughan** [comment]
Negative `n` is an error as of https://github.com/r-lib/vctrs/pull/1523

Otherwise we closed PR #1036 which was going to alter the signature of this function. We determined there that it is probably not worth it to do this anymore
